store/tikv/2pc.go

```diff
@@ -702,6 +705,13 @@ func (c *twoPhaseCommitter) pessimisticLockSingleBatch(bo *Backoffer, batch batc
 		}
 		var locks []*Lock
 		for _, keyErr := range keyErrs {
+			// Check lock conflict error for nowait, if nowait set and key locked by others
+			// report error immediately and do no resolve locks any more
+			if c.lockNoWait {
+				if keyErr.GetLocked() != nil {
```

Comment:
can this lock be a `committed` or `rollbacked` lock in pessimistic txn and need check lock state? I'm not familiar with this :D